openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -408,34 +417,66 @@ def _verify_user(request, response, userid_in_session):
             return
 
         if hasattr(request, 'safe_cookie_verified_user_id'):
+            # The user at response time (and in the session) is expected to be None when the user
+            # is logging out so do not treat it as a mismatch
+            is_request_response_mismatch = request.safe_cookie_verified_user_id != request.user.id\
+                and request.user.id is not None
+            is_request_session_mismatch = request.safe_cookie_verified_user_id != userid_in_session\
+                and userid_in_session is not None
+
             if hasattr(request.user, 'real_user'):
                 # If a view overrode the request.user with a masqueraded user, this will
-                #   revert/clean-up that change during response processing.
+                # revert/clean-up that change during response processing.
                 request.user = request.user.real_user
-            # The user at response time is expected to be None when the user
-            # is logging out.  We won't log that.
-            if request.safe_cookie_verified_user_id != request.user.id and request.user.id is not None:
-                log.warning(
-                    (
-                        "SafeCookieData user at request '{}' does not match user at response: '{}' "
-                        "for request path '{}'"
-                    ).format(  # pylint: disable=logging-format-interpolation
-                        request.safe_cookie_verified_user_id, request.user.id, request.path,
-                    ),
-                )
-                set_custom_attribute("safe_sessions.user_mismatch", "request-response-mismatch")
-            # The user session at response time is expected to be None when the user
-            # is logging out.  We won't log that.
-            if request.safe_cookie_verified_user_id != userid_in_session and userid_in_session is not None:
-                log.warning(
-                    (
-                        "SafeCookieData user at request '{}' does not match user in session: '{}' "
-                        "for request path '{}'"
-                    ).format(  # pylint: disable=logging-format-interpolation
-                        request.safe_cookie_verified_user_id, userid_in_session, request.path,
-                    ),
-                )
-                set_custom_attribute("safe_sessions.user_mismatch", "request-session-mismatch")
+
+            session_at_response = 'None'
+            if hasattr(request, 'session') and hasattr(request.session, 'session_key'):
+                session_at_response = request.session.session_key
+
+            if is_request_response_mismatch or is_request_session_mismatch:
+                if not cache.touch(USER_MISMATCH_CACHE_KEY, 300):
+                    cache.set(USER_MISMATCH_CACHE_KEY, True, 300)
```

Comment:
What's the benefit of this touch/set approach?

Reply:
It felt conceptually closer to "extend the timeout period for every incident" but I'm not married to it